File: src/main/java/number_lines_jump/NumberLinesJump.java
package number_lines_jump;

public class NumberLinesJump {
    public void numberLinesJump() {

        int x1 = 0;
        int v1 = 3;

        int x2 = 4;
        int v2 = 2;

        for (int i = 1; true; i++) {
            if (x1 + v1 * i == x2 + v2 * i) {
                System.out.println("YES");
                break;
            } else {
                System.out.println("NO");
            }
        }
    }
}

File: src/main/kotlin/number_lines_jump/numberLinesJump.kt
package number_lines_jump

fun numberLinesJump() {
    val x1 = 0
    val v1 = 3
    val x2 = 4
    val v2 = 2

    for (i in 1..10000) {
        if (x1 + v1 * i == x2 + v2 * i) {
            println("YES")
            break
        } else {
            println("NO")
        }
    }
}